lib/media/media_source_engine.js

```
@@ -1690,11 +1690,20 @@ shaka.media.MediaSourceEngine = class {
    */
   async setDuration(duration) {
     await this.enqueueBlockingOperation_(() => {
-      // Reducing the duration causes the MSE removal algorithm to run, which
-      // triggers an 'updateend' event to fire.  To handle this scenario, we
-      // have to insert a dummy operation into the beginning of each queue,
-      // which the 'updateend' handler will remove.
-      if (duration < this.mediaSource_.duration) {
+      // https://www.w3.org/TR/media-source-2/#duration-change-algorithm
+      // "Duration reductions that would truncate currently buffered media
+      // are disallowed.
+      // When truncation is necessary, use remove() to reduce the buffered
+      // range before updating duration."
+      // But in some platforms, truncating the duration causes the
```

Was there a faulty assumption behind PR #4483? I was not under the impression this was a platform-specific issue.

Anyway, if this is a platform-specific issue, could we perhaps set the new `durationReductionEmitsUpdateEnd` config value to true by default for the appropriate platform or platforms? I'm worried about what will happen if we change our default behavior in a way that might break existing apps, and hide the old behavior behind a newly added configuration.

---

@srikieonline any comment?

---

I've already pushed a change to set the default value to true. So it wont break any platform.,